MovieApp/MovieApp/App/Coordinator/AppRouter.swift

```diff
@@ -22,7 +22,7 @@ final class AppRouter {
     }
     
     func showDetailScreen(for movieId: Int) {
-        let movieDetailPresenter = MovieDetailPresenter()
+        let movieDetailPresenter = MovieDetailPresenter(useCase: moviesUseCase, router: self)
         let vc = MovieDetailViewController(presenter: movieDetailPresenter, withMovieId: movieId)
```

Comment:
Generally, if there's an ID passed, it's passed to the presenter since it's responsible for fetching all the data from the UC.

Reply:
[Changed](https://github.com/lihter/Movie-App/pull/30/commits/cca37b805fab60d09dcbfc057d25f02fa279dbbc#diff-b8e8fb675a7084dca12ae6abb54b13ee1d7293192f1e1e75308b4dca15ea5465R25-R26)